py/torch_tensorrt/fx/passes/lower_basic_pass_aten.py

```diff
@@ -412,33 +412,50 @@ def compose_bmm(
     modified = False
     for n in module.graph.nodes:
         if n.op == "call_function" and n.target in (torch.ops.aten.bmm.default,):
-            modified = True
+            modified = False
             node = n
             input_n = node.all_input_nodes[0]
             other_n = node.all_input_nodes[1]
+
+            # If no input nodes are available, the bmm argument itself could be an input
+            # Alternatively, if the node has no users, it can be eliminated
+            if len(input_n.all_input_nodes) == 0 or len(node.users) == 0:
```

Comment:
do we have such case which has no inputs to the bmm?

Reply:
Yes, the case in #1789, which uses this model, has `len(input_n.all_input_nodes) == 0`:
```python
def forward(self, x, y):
    out = torch.bmm(x, y)
    return out
```
 The test case shows this behavior too.

Reply:
I did an experiment. 1) Looks like `bmm` is not decomposed anymore.  2) bmm's all_input_nodes is not 0
<img width="1046" alt="image" src="https://user-images.githubusercontent.com/6955737/232973078-c1979886-c2bd-4d8b-95c1-114eeb385f30.png">

Reply:
@frank-wei I see, thanks for the additional details. The reason I included that test case was for the case where we take the model in question and apply the `compose_bmm` lowering pass to it, in which case we see the failure. Specifically, in the above example, add the following lines:
```python
from torch_tensorrt.fx.passes.lower_basic_pass_aten import compose_bmm
composed_module = compose_bmm(gm[0])
out = composed_module.graph_module(*input2)
```
When doing so, the following error is elicited:
```python
  File "test.py", line 17, in <module>
    composed_module = compose_bmm(gm[0])
  File "~/TensorRT/py/torch_tensorrt/fx/passes/lower_basic_pass_aten.py", line 409, in compose_bmm
    input_input_n = input_n.all_input_nodes[0]
IndexError: list index out of range
```